### ISSUE TITLE
Rest API equivalent of setCustomUserClaims as I can't use Firebase Admin SDK in non-nodejs runtime?

### ISSUE THREAD

**Issue by GorvGoyl** [firebase/firebase-admin-node#2122]
Firebase admin SDK is not supported on [Vercel edge functions](https://vercel.com/docs/concepts/functions/edge-functions/limitations#most-node.js-apis-are-not-available) so how can I set [custom user claims](https://firebase.google.com/docs/auth/admin/custom-claims#set_and_validate_custom_user_claims_via_the_admin_sdk) in Firebase auth user if I have uid and id token?

**Comment by lahirumaramba** [comment]
You are correct, `firebase-admin` currently requires a full Node.js runtime. The SDK uses the following REST API to set custom claims.

REST API: https://cloud.google.com/identity-platform/docs/reference/rest/v1/projects.tenants.accounts/update

https://github.com/firebase/firebase-admin-node/blob/5533a94b0519e67edbc99833353f73bdefdb345d/src/auth/auth-api-request.ts#L1336

You will need to:

- Authenticate as a user who is at least an editor on the project.
- Build a request with a `customAttributes` property for the new claims.
- Call the `accounts:update` auth REST endpoint.

Please refer to the [REST API reference docs](https://cloud.google.com/identity-platform/docs/reference/rest/v1/projects.tenants.accounts/update) and the SDK implementation for more details. Additionally, here is a similar answer on SO: https://stackoverflow.com/a/69451750/1526201

**Comment by awinogrodzki** [comment]
Just FYI: you can now `setCustomUserClaims` in [Edge Runtime](https://nextjs.org/docs/api-reference/edge-runtime) using `next-firebase-auth-edge` lib 

https://github.com/awinogrodzki/next-firebase-auth-edge/issues/32
